packages/components/src/snackbar/style.scss

```diff
@@ -29,13 +29,14 @@
 	}
 
 	.components-snackbar__content-with-icon {
-		margin-left: $grid-unit-30;
+		position: relative;
+		padding-left: $grid-unit-30;
 	}
 
 	.components-snackbar__icon {
 		position: absolute;
-		top: 24px;
-		left: 28px;
+		left: $grid-unit-10 * -1;
+		top: -3px;
```

Comment:
Is the icon supposed to be top aligned when the snackbar text breaks on 2+ lines?

![Screenshot 2024-01-02 at 18 04 49](https://github.com/WordPress/gutenberg/assets/1083581/e80a324a-4a16-43fe-86c8-c6727c022968)

More in general, we should probably switch to a more generic approach to vertically centering the icon that doesn't rely on an exact px value, since it would make the alignment more resilient to things like font changes, etc.

Reply:
> Is the icon supposed to be top aligned when the snackbar text breaks on 2+ lines?

Correct. It matches the alignment of the actions and dismiss button as well.

<img src="https://private-user-images.githubusercontent.com/555336/292870477-6b46c69f-cbe8-4071-ae73-f2ece83e8d37.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQyMTc5ODUsIm5iZiI6MTcwNDIxNzY4NSwicGF0aCI6Ii81NTUzMzYvMjkyODcwNDc3LTZiNDZjNjlmLWNiZTgtNDA3MS1hZTczLWYyZWNlODNlOGQzNy5QTkc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEwMlQxNzQ4MDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iNTI3NjQwNTZmNGM3NjM1ZDYyNTBmM2RmYzk5MDIxYjcyOGM3OTUyY2U2OTdlYzkyNWU0ODg4MGQ0NTUzMGE5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.jekM-B66orrl_JWnvXryvCIeXU6am7EjsPozv-zrQkg" alt="Snackbar with actions and dismiss button" width="400">

> More in general, we should probably switch to a more generic approach to vertically centering the icon that doesn't rely on an exact px value, since it would make the alignment more resilient to things like font changes, etc.

Added a few tweaks to avoid this in 312752d 👍